src/units/combine.rs

```diff
@@ -306,3 +175,69 @@ impl metrics::Source for AnyMetrics {
     }
 }
 
+
+//============ Tests =========================================================
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use futures::join;
+    use tokio::runtime;
+    use crate::{test, units};
+    use crate::manager::Manager;
+    use crate::payload::testrig;
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn wake_up_again() {
+        use crate::comms::UnitStatus::*;
+
+        let mut manager = Manager::new();
+
+        let (u1, u2, u3, mut t) = manager.add_units(
+            &runtime::Handle::current(),
+            |units, targets| {
+                let (u, u1c) = test::Unit::new();
+                units.insert("u1", u);
+                let (u, u2c) = test::Unit::new();
+                units.insert("u2", u);
+                let (u, u3c) = test::Unit::new();
+                units.insert("u3", u);
+
+                units.insert("any", units::Unit::Any(Any {
+                    sources: vec!["u1".into(), "u2".into(), "u3".into()],
+                    random: false
+                }));
+
+                let (t, tc) = test::Target::new("any");
+                targets.insert("t", t);
+
+                (u1c, u2c, u3c, tc)
+            }
+        ).unwrap();
+
+        // Set all units to stalled, check that the target goes stalled.
+        join!(u1.status(Stalled), u2.status(Stalled), u3.status(Stalled));
+        assert_eq!(t.recv().await, Err(Stalled));
+
+        // Set one unit to healthy.
+        u2.data(testrig::update(&[1])).await;
+        u2.status(Healthy).await;
+        assert_eq!(t.recv().await, Err(Healthy));
```

Comment:
Why are the receives in reverse order of the sends?

Reply:
Good question! I need to investigate.

Reply:
Ah, so this because there’s the any unit in between. It only reacts when the status goes to healthy and then send a status update followed by the data update. This sequencing is kind of icky, so I suppose we should rewrite that into allowing both a data and a status update in one message. I’d like to do that in a follow-up PR, though?